hollow/src/main/java/com/netflix/hollow/api/client/HollowDataHolder.java

```diff
@@ -111,18 +112,21 @@ void update(HollowUpdatePlan updatePlan, HollowConsumer.RefreshListener[] refres
         // such as when a new delta is published.
         // Note that a refresh listener may also induce a failed transition, likely unknowingly,
         // by throwing an exception.
-        if(doubleSnapshotConfig.allowDoubleSnapshot() && failedTransitionTracker.anyTransitionWasFailed(updatePlan)) {
+        if (doubleSnapshotConfig.allowDoubleSnapshot() && failedTransitionTracker.anyTransitionWasFailed(updatePlan)) {
             throw new RuntimeException("Update plan contains known failing transition!");
         }
 
-        if(updatePlan.isSnapshotPlan())
-            applySnapshotPlan(updatePlan, refreshListeners);
-        else
+        if (updatePlan.isSnapshotPlan()) {
+            applySnapshotPlan(updatePlan, refreshListeners, apiInitCallback);
+        } else {
             applyDeltaOnlyPlan(updatePlan, refreshListeners);
+        }
     }
 
-    private void applySnapshotPlan(HollowUpdatePlan updatePlan, HollowConsumer.RefreshListener[] refreshListeners) throws Throwable {
-        applySnapshotTransition(updatePlan.getSnapshotTransition(), refreshListeners);
```

Comment:
A few lines below this, `refreshListener.snapshotUpdateOccurred()` is called, might this be a problem if the `HollowDataHolder` is not assigned in the updater yet?

Reply:
Yes indeed, that would be a problem. This PR doesn't change that though (more detail in the response to your other comment)